src/App/Services/Tamara_Widget.php

```diff
@@ -89,4 +100,42 @@ public function fetch_tamara_cart_widget( $data = [] ) {
 			]
 		);
 	}
+
+	/**
+	 * @throws \Exception
+	 */
+	public function fetch_tamara_checkout_widget(): string {
+		$widget_inline_type = 3;
+		$cart_amount = WC_Order_Helper::define_total_amount_to_calculate( WC()->cart->total );
+		$description = Tamara_Checkout_WP_Plugin::wp_app_instance()->view(
+			'blocks/tamara-widget',
+			[
+				'widget_inline_type' => $widget_inline_type,
+				'widget_amount' => $cart_amount,
+			]
+		);
+
+		$description .= $this->populate_default_description_text_on_checkout();
+		return $description;
+	}
+
+	/**
+	 * Populate tamara default description text on checkout
+	 *
+	 * @throws \Exception
+	 */
+	protected function populate_default_description_text_on_checkout(): string {
+		$allowed_countries_text = 'Saudi Arabia, Kuwait, UAE and Qatar only.<br>';
+		$description = $this->_t( sprintf( '*Exclusive for shoppers in %s', $allowed_countries_text ) );
+		if ( ! Tamara_Checkout_WP_Plugin::wp_app_instance()->get_tamara_gateway_service()->get_settings()->is_live_mode ) {
+			$description .= '<br/>' . $this->_t(
```

Comment:
sprintf should work before `_t`, but you don't have anything to replace, why do you need sprintf. BTW, we need to use the correct link to the guides